integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextbest/MessageOutController.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    @RequestMapping(value = "/out/messages", method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity createResource(
+            @RequestParam("receiverId") String receiverId,
+            @RequestParam("messagetypeId") String messagetypeId,
+            @RequestParam(value = "senderId", required = false) String senderId,
+            @RequestParam(value = "conversationId", required = false) String conversationId) throws URISyntaxException {
+
+        if (isNullOrEmpty(receiverId)) {
+            return ResponseEntity.badRequest().body("Required String parameter \'receiverId\' is not present");
+        }
+        if (isNullOrEmpty(messagetypeId)) {
+            return ResponseEntity.badRequest().body("Required String parameter \'messagetypeId\' is not present");
+        }
+
+        String sender;
+        if (isNullOrEmpty(senderId)) {
+            sender = props.getOrg().getNumber();
+        } else {
+            sender = senderId;
+        }
+
+        OutgoingConversationResource conversationResource;
+        if (isNullOrEmpty(conversationId)) {
+            conversationResource = OutgoingConversationResource.of(sender, receiverId, messagetypeId);
+        } else {
+            conversationResource = repo.findOne(conversationId);
+            if (conversationResource == null) {
+                conversationResource = OutgoingConversationResource.of(conversationId, sender, receiverId,
+                        messagetypeId);
+            }
+        }
+        repo.save(conversationResource);
+        return ResponseEntity.ok(conversationResource);
+    }
+
+    @RequestMapping(value = "/out/messages/{conversationId}", method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity uploadFiles(
+            @PathVariable("conversationId") String conversationId,
+            MultipartHttpServletRequest request) {
+
+        OutgoingConversationResource conversationResource = repo.findOne(conversationId);
+        if (conversationResource == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("No conversation with supplied id found");
+        }
+
+        ArrayList<String> files = Lists.newArrayList(request.getFileNames());
+        for (String f : files) {
+            MultipartFile file = request.getFile(f);
+            log.info("Key: "+file.getName());
+            log.info("Name: "+file.getOriginalFilename());
+            log.info("Content-Type: "+file.getContentType());
+            log.info("Size (bytes): "+file.getSize());
+
+            String filedir = props.getNextbest().getFiledir();
+            if (!filedir.endsWith("/")) {
+                filedir = filedir+"/";
+            }
+            filedir = filedir+conversationId+"/";
+            File localFile = new File(filedir+file.getOriginalFilename());
+            localFile.getParentFile().mkdirs();
+
+            try {
+                FileOutputStream os = new FileOutputStream(localFile);
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Close this "FileOutputStream". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS2095)